ansible/roles/ocp-workload-3scale-demo/tasks/config.yml

```diff
@@ -10,9 +10,22 @@
   retries: 15
   delay: 60
 
+- name: Retrieve SSO admin credentials
+  k8s_facts:
+    kind: secret
+    name: credential-sso
+    namespace: '{{sso_project}}'
+  register: _sso_credentials
+
+- debug: var=_sso_credentials
+
+- set_fact:
+    _sso_admin_user: '{{ _sso_credentials.resources[0].data.ADMIN_USERNAME | b64decode }}'
+    _sso_admin_password: '{{ _sso_credentials.resources[0].data.ADMIN_PASSWORD | b64decode }}'
+
 - name: Get SSO token
-  shell: "curl --insecure -X POST 'https://secure-sso-{{sso_project}}.{{ocp_apps_domain}}/auth/realms/master/protocol/openid-connect/token' \
-  -H 'Content-Type: application/x-www-form-urlencoded' -d 'username=keyadmin' -d 'password=keypassword' -d 'grant_type=password' -d 'client_id=admin-cli'"
+  shell: "curl --insecure -X POST 'https://keycloak-{{sso_project}}.{{ocp_apps_domain}}/auth/realms/master/protocol/openid-connect/token' \
+  -H 'Content-Type: application/x-www-form-urlencoded' -d 'username={{_sso_admin_user}}' -d 'password={{_sso_admin_password}}' -d 'grant_type=password' -d 'client_id=admin-cli'"
```

Comment:
Please reformat for readability, and switch from `shell` to `command` because no use of shell functionality... and best to use `quote` filter rather than relying on the password not having an `'` in it.
```
- name: Get SSO token
  command: >-
    curl --insecure -X POST 'https://keycloak-{{sso_project}}.{{ocp_apps_domain}}/auth/realms/master/protocol/openid-connect/token'
    -H 'Content-Type: application/x-www-form-urlencoded'
    -d username={{ _sso_admin_user | quote }}
    -d password={{_sso_admin_password | quote }}
    -d grant_type=password
    -d client_id=admin-cli
  register: token_text

- set_fact: TKN={{ (token_text.stdout | from_json).access_token }}
```
Or... even better, consider switching to use of the `uri` module.
```
- name: Get SSO token
  uri:
    method: POST
    url: https://keycloak-{{sso_project}}.{{ocp_apps_domain}}/auth/realms/master/protocol/openid-connect/token
    validate_certs: false
    body_format: form-urlencoded
    body:
      username: "{{ _sso_admin_user }}"
      password: "{{_sso_admin_password }}"
      grant_type: password
      client_id: admin-cli
  register: r_get_token

- name: Set TKN
  set_fact: TKN={{ r_get_token.json.access_token }}
```